diff -aburN a/drivers/misc/akm8963.c b/drivers/misc/akm8963.c
--- a/drivers/misc/akm8963.c	2014-09-25 23:49:09.000000000 -0300
+++ b/drivers/misc/akm8963.c	2015-11-19 11:19:47.216000000 -0200
@@ -816,7 +816,7 @@
 	flag = ((akm->enable_flag >> pos) & 1);
 	mutex_unlock(&akm->val_mutex);
 
-	return snprintf(buf,sizeof(buf), "%d\n", flag);
+	return sprintf(buf, "%d\n", flag);
 }
 
 static ssize_t akm8963_sysfs_enable_store(
@@ -894,7 +894,7 @@
 	val = akm->delay[pos];
 	mutex_unlock(&akm->val_mutex);
 
-	return snprintf(buf,sizeof(buf), "%lld\n", val);
+	return sprintf(buf, "%lld\n", val);
 }
 
 static ssize_t akm8963_sysfs_delay_store(
@@ -1019,7 +1019,7 @@
 	memcpy(&rbuf, akm->sense_data, sizeof(rbuf));
 	mutex_unlock(&akm->sensor_mutex);
 
-	return snprintf(buf,sizeof(buf),
+	return sprintf(buf,
 		"0x%02X,0x%02X,0x%02X,0x%02X,"
 		"0x%02X,0x%02X,0x%02X,0x%02X\n",
 		rbuf[0],rbuf[1],rbuf[2],rbuf[3],
@@ -1046,7 +1046,7 @@
 	if (err < 0)
 		return err;
 
-	return snprintf(buf, sizeof(buf),"0x%02X,0x%02X,0x%02X\n",
+	return sprintf(buf, "0x%02X,0x%02X,0x%02X\n",
 		asa[0], asa[1], asa[2]);
 }
 #endif
diff -aburN a/net/bluetooth/hci_conn.c b/net/bluetooth/hci_conn.c
--- a/net/bluetooth/hci_conn.c	2014-09-25 23:49:12.000000000 -0300
+++ b/net/bluetooth/hci_conn.c	2015-11-18 23:15:54.276000000 -0200
@@ -404,7 +404,7 @@
 	memset(&cp, 0, sizeof(cp));
 
 	cp.handle = cpu_to_le16(conn->handle);
-	memcpy(cp.ltk, ltk, sizeof(ltk));
+	memcpy(cp.ltk, ltk, sizeof(cp.ltk));
 
 	hci_send_cmd(hdev, HCI_OP_LE_LTK_REPLY, sizeof(cp), &cp);
 }
